ext/liquid_c/vm_assembler.c

```diff
@@ -73,6 +74,113 @@ void vm_assembler_gc_mark(vm_assembler_t *code)
     }
 }
 
+VALUE vm_assembler_disassemble(vm_assembler_t *code)
+{
+    const size_t *const_ptr = (size_t *)code->constants.data;
+    const uint8_t *start_ip = code->instructions.data;
+    const uint8_t *ip = start_ip;
+    const uint8_t *end_ip = code->instructions.data_end;
+    VALUE output = rb_str_buf_new(32);
+    while (ip < end_ip) {
+        rb_str_catf(output, "0x%04lx: ", ip - start_ip);
+        switch (*ip) {
+            case OP_LEAVE:
+                rb_str_catf(output, "leave\n");
+                break;
+
+            case OP_POP_WRITE:
+                rb_str_catf(output, "pop_write\n");
+                break;
+
+            case OP_PUSH_NIL:
+                rb_str_catf(output, "push_nil\n");
+                break;
+
+            case OP_PUSH_TRUE:
+                rb_str_catf(output, "push_true\n");
+                break;
+
+            case OP_PUSH_FALSE:
+                rb_str_catf(output, "push_false\n");
+                break;
+
+            case OP_FIND_VAR:
+                rb_str_catf(output, "find_var\n");
+                break;
+
+            case OP_LOOKUP_KEY:
+                rb_str_catf(output, "lookup_key\n");
+                break;
+
+            case OP_NEW_INT_RANGE:
+                rb_str_catf(output, "new_int_range\n");
+                break;
+
+            case OP_HASH_NEW:
+                rb_str_catf(output, "hash_new(%u)\n", ip[1]);
+                break;
+
+            case OP_PUSH_INT8:
+                rb_str_catf(output, "push_int8(%u)\n", ip[1]);
+                break;
+
+            case OP_PUSH_INT16:
+            {
+                int num = *(int8_t *)ip++;
+                num = (num << 8) | ip[1];
```

Comment:
`OP_PUSH_INT16` is broken.
```suggestion
                int num = (ip[1] << 8) | ip[2];
```

Reply:
You're right.  I pushed a commit to fix it along with a test for it.